lib/index.js

```diff
@@ -94,18 +94,18 @@ module.exports = function (
     const args = nodeArgs.slice();
 
     args.push(`--require=${resolveMain(localPath('wrap'))}`);
-    const loaderPath = semver.satisfies(process.version, '<16.12.0')
-      ? 'loader-legacy.mjs'
-      : 'loader.mjs';
-    const loader = pathToFileURL(resolveMain(localPath(loaderPath)));
+
     if (semver.satisfies(process.version, '<12.17.0')) {
       args.push('--experimental-modules');
     }
-    if (semver.satisfies(process.version, '<12.11.1')) {
-      args.push(`--loader=${loader.href}`);
-    } else {
-      args.push(`--experimental-loader=${loader.href}`);
-    }
+
+    const legacyPrefix = semver.satisfies(process.version, '<16.12.0') ? 'legacy-' : '';
+    const loaderURL = pathToFileURL(resolveMain(localPath(`${legacyPrefix}loader.mjs`)));
+    const experimentalPrefix = semver.satisfies(process.version, '>=12.11.1')
+      ? 'experimental-'
+      : '';
+
+    args.push(`--${experimentalPrefix}loader=${loaderURL.href}`);
 
```

Comment:
I think the code golf here might be overfitting to what the loader options currently are. I'm concerned that when the loader option's name inevitably changes again in the future that all of this string concatenation won't be a maintainable approach.

Reply:
Sorry, I just get annoyed when both sides of an if statement have the same function call with different arguments.